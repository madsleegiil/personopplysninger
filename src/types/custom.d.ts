declare module "*.svg" {
  const content: any;
  export default content;
}

declare module "*.png" {
  const content: any;
  export default content;
}

declare module "@navikt/arbeidsforhold" {
  const content: any;
  export default content;
}
