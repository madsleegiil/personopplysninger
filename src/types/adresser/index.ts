import { Boadresse } from "./boadresse";
import { Postadresse } from "./postadresse";
import { Tilleggsadresse } from "./tilleggsadresse";
import { UtenlandskAdresse } from "./utenlandskadresse";
import { Kontaktadresse } from "./kontaktadresse";

export interface GeografiskTilknytning {
  bydel?: string;
  datoFraOgMed?: string;
  kommune?: string;
  enhet?: string;
  land?: string;
}

export interface Adresser {
  boadresse?: Boadresse;
  geografiskTilknytning?: GeografiskTilknytning;
  postadresse?: Postadresse;
  prioritertAdresse?: string;
  kontaktadresse: Kontaktadresse;
  datakilder?: {}[];
}
