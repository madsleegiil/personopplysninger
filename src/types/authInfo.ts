export type AuthInfo =
  | {
      authenticated: false;
    }
  | {
      authenticated: true;
      name: string;
      securityLevel: string;
    };

export interface AuthOidc {
  authenticated: true | false;
}
