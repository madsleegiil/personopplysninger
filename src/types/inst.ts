export interface InstInnslag {
  organisasjonsnummer: string;
  institusjonsnavn: string;
  institusjonstype: string;
  kategori: string;
  startdato: string;
  faktiskSluttdato: string;
  registreringstidspunkt: string;
  maskineltAvsluttet: boolean;
}

export type InstInfo = InstInnslag[];
