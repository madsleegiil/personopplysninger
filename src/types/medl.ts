export interface MedlInnslag {
  unntakId: number;
  ident: string;
  fraOgMed: string;
  tilOgMed: string;
  status: "Avvist" | "Gyldig" | "Uavklart";
  statusaarsak: string;
  dekning: string;
  helsedel: boolean;
  medlem: boolean;
  lovvalgsland: string;
  lovvalg: string;
  grunnlag: string;
}

export type MedlInfo = MedlInnslag[];
