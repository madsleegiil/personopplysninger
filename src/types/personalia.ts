export interface Tlfnr {
  jobb?: string;
  mobil?: string;
  privat?: string;
}

export interface Personalia {
  fornavn?: string;
  etternavn?: string;
  personident?: {
    verdi: string;
    type: string;
  };
  kontonr?: string;
  tlfnr?: Tlfnr;
  spraak?: string;
  epostadr?: string;
  personstatus?: string;
  statsborgerskap?: string;
  foedested?: string;
  sivilstand?: string;
  kjoenn?: string;
  datakilder?: {}[];
}
