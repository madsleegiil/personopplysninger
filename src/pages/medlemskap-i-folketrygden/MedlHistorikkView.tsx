import React from "react";
import { MedlInfo } from "types/medl";
import { AlertStripeInfo } from "nav-frontend-alertstriper";
import { FormattedMessage, useIntl } from "react-intl";
import { Normaltekst, Systemtittel, Undertittel } from "nav-frontend-typografi";
import Lenke from "nav-frontend-lenker";
import Panel from "./Panel";
import Check from "../../assets/img/Check.svg";

interface Props {
  medlInfo: MedlInfo;
}

const MedlHistorikkView = (props: Props) => {
  const { medlInfo } = props;
  const { locale } = useIntl();
  const perioder = medlInfo?.perioder || [];

  const medMedlemskap = perioder.filter(
    (periode) => periode.medlem && !periode.studieinformasjon
  );
  const utenMedlemskap = perioder.filter((periode) => !periode.medlem);
  const fraLanekassen = perioder.filter((periode) => periode.studieinformasjon);
  const ingenPerioder =
    medMedlemskap.length === 0 &&
    utenMedlemskap.length === 0 &&
    fraLanekassen.length === 0;

  const lenkerTilFolketrygden: { [key: string]: string } = {
    nb: `/no/person/flere-tema/arbeid-og-opphold-i-norge/relatert-informasjon/medlemskap-i-folketrygden`,
    nn: `/no/person/flere-tema/arbeid-og-opphold-i-norge/nynorsk/medlemskap-i-folketrygda`,
    en: `/en/home/rules-and-regulations/membership-of-the-national-insurance-scheme`,
  };

  return (
    <div className="medl__tabs-innhold">
      <div className={"medl__space"}>
        <Normaltekst>
          <FormattedMessage id={`medl.oversikt.ingress`} />
        </Normaltekst>
      </div>
      <div className={"medl__space"}>
        <Checkbox text={`medl.oversikt.1`} />
      </div>
      <div className={"medl__space"}>
        <Checkbox text={`medl.oversikt.2`} />
      </div>

      <div className={"medl__section"}>
        <Undertittel>
          <FormattedMessage id={"medl.situasjoner.tittel"} />
        </Undertittel>
      </div>
      <div className={"medl__space"}>
        <Checkbox text={`medl.situasjoner.1`} />
      </div>
      <div className={"medl__space"}>
        <Checkbox text={`medl.situasjoner.2`} />
      </div>
      <div className={"medl__space"}>
        <Checkbox text={`medl.situasjoner.3`} />
      </div>

      <div className={"medl__section"}>
        <Undertittel>
          <FormattedMessage id={"medl.folketrygden.tittel"} />
        </Undertittel>
      </div>
      <div className={"medl__space"}>
        <Normaltekst>
          <FormattedMessage
            id={`medl.folketrygden.ingress`}
            values={{
              lenkeTilFolketrygden: (text: string) => (
                <Lenke href={lenkerTilFolketrygden[locale]}>{text}</Lenke>
              ),
            }}
          />
        </Normaltekst>
      </div>
      <div className={"medl__space"}>
        <Normaltekst>
          <FormattedMessage id={`medl.folketrygden.lanekassen`} />
        </Normaltekst>
      </div>

      <div className={"medl__section"}>
        <Systemtittel>
          <FormattedMessage id={"medl.perioder.tittel"} />
        </Systemtittel>
      </div>
      <div className={"medl__space"}>
        <AlertStripeInfo>
          <FormattedMessage id={"medl.perioder.alert"} />
        </AlertStripeInfo>
      </div>
      {medMedlemskap.length > 0 ? (
        <Panel
          tittelId={"medl.medlemskap.med.tittel"}
          tittelIdIngress={`medl.medlemskap.med.ingress`}
          perioder={medMedlemskap}
        />
      ) : null}
      {utenMedlemskap.length > 0 ? (
        <Panel
          tittelId={"medl.medlemskap.uten.tittel"}
          tittelIdIngress={`medl.medlemskap.uten.ingress`}
          perioder={utenMedlemskap}
        />
      ) : null}
      {fraLanekassen.length > 0 ? (
        <Panel
          tittelId={"medl.medlemskap.lanekassen.tittel"}
          tittelIdIngress={`medl.medlemskap.lanekassen.ingress`}
          perioder={fraLanekassen}
        />
      ) : null}
      {ingenPerioder && (
        <div className={"medl__space"}>
          <AlertStripeInfo>
            <FormattedMessage id={"medl.ingendata"} />
          </AlertStripeInfo>
        </div>
      )}
    </div>
  );
};

const Checkbox = (props: { text: string }) => (
  <div className="medl__checkbox">
    <img className={"medl__kalender"} src={Check} alt="Sjekkboksikon" />
    <Normaltekst>
      <FormattedMessage id={props.text} />
    </Normaltekst>
  </div>
);

export default MedlHistorikkView;
