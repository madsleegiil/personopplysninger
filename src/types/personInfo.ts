import { Personalia } from "./personalia";
import { Adresser } from "./adresser";

export interface PersonInfo {
  personalia?: Personalia;
  adresser?: Adresser;
}
