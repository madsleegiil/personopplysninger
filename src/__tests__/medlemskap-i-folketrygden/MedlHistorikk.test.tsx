import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { IntlProvider } from "react-intl";
import nbMessages from "text/nb";
import MedlHistorikkView from "../../pages/medlemskap-i-folketrygden/MedlHistorikkView";
import medlInfo from "../../clients/apiMock/app/fetch/medl-info.json";
import { MedlInfo } from "../../types/medl";

jest.mock("react-router-dom", () => ({
  Link: () => <div />,
  useLocation: () => ({
    pathname: "pathname",
  }),
}));
jest.mock("nav-frontend-js-utils", () => ({
  ...jest.requireActual("nav-frontend-js-utils"),
  guid: () => "Mocked UUID",
}));

jest.mock("nav-frontend-hjelpetekst");

describe("MedlHistorikkView", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <IntlProvider locale={"nb"} messages={nbMessages}>
          <MedlHistorikkView medlInfo={medlInfo as unknown as MedlInfo} />
        </IntlProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
