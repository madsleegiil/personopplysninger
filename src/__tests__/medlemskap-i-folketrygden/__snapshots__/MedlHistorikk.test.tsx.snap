// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MedlHistorikkView renders correctly 1`] = `
<DocumentFragment>
  <div
    class="medl__tabs-innhold"
  >
    <div
      class="medl__space"
    >
      <p
        class="typo-normal"
      >
        Hvis du bor og jobber i Norge, er du vanligvis medlem i folketrygden. Perioder med denne type medlemskap vil ikke vises i oversikten. Oversikten er kun aktuell for deg som har
      </p>
    </div>
    <div
      class="medl__space"
    >
      <div
        class="medl__checkbox"
      >
        <img
          alt="Sjekkboksikon"
          class="medl__kalender"
          src="Check.svg"
        />
        <p
          class="typo-normal"
        >
          bodd, jobbet eller studert i utlandet, og du har søkt om medlemskap i folketrygden og fått bekreftet det.
        </p>
      </div>
    </div>
    <div
      class="medl__space"
    >
      <div
        class="medl__checkbox"
      >
        <img
          alt="Sjekkboksikon"
          class="medl__kalender"
          src="Check.svg"
        />
        <p
          class="typo-normal"
        >
          jobbet i Norge, men var med i trygdeordningen i et annet land.
        </p>
      </div>
    </div>
    <div
      class="medl__section"
    >
      <h2
        class="typo-undertittel"
      >
        Ulike situasjoner
      </h2>
    </div>
    <div
      class="medl__space"
    >
      <div
        class="medl__checkbox"
      >
        <img
          alt="Sjekkboksikon"
          class="medl__kalender"
          src="Check.svg"
        />
        <p
          class="typo-normal"
        >
          Flytter du til utlandet for å bo eller jobbe, avhenger medlemskap i folketrygden blant annet av hva du gjør og varigheten på oppholdet.
        </p>
      </div>
    </div>
    <div
      class="medl__space"
    >
      <div
        class="medl__checkbox"
      >
        <img
          alt="Sjekkboksikon"
          class="medl__kalender"
          src="Check.svg"
        />
        <p
          class="typo-normal"
        >
          Norske studenter i utlandet vil vanligvis være pliktige medlemmer i folketrygden, hvis de får støtte fra Lånekassen for å studere på heltid ved et utenlandsk universitet eller en høyskole.
        </p>
      </div>
    </div>
    <div
      class="medl__space"
    >
      <div
        class="medl__checkbox"
      >
        <img
          alt="Sjekkboksikon"
          class="medl__kalender"
          src="Check.svg"
        />
        <p
          class="typo-normal"
        >
          Er du bosatt i et annet land, men jobber helt eller delvis i Norge, kan du være med i trygdeordningen i bostedslandet ditt og være unntatt fra medlemskap i folketrygden.
        </p>
      </div>
    </div>
    <div
      class="medl__section"
    >
      <h2
        class="typo-undertittel"
      >
        Hva betyr medlemskap i folketrygden?
      </h2>
    </div>
    <div
      class="medl__space"
    >
      <p
        class="typo-normal"
      >
        Medlemskap i folketrygden betyr at du har rettigheter og plikter i folketrygden. 
        <a
          class="lenke"
          href="/no/person/flere-tema/arbeid-og-opphold-i-norge/relatert-informasjon/medlemskap-i-folketrygden"
        >
           Les om medlemskap i folketrygden
        </a>
        . Kontakt NAV hvis du har spørsmål om perioder med eller uten medlemskap i folketrygden, telefon (+47) 21 07 37 00.
      </p>
    </div>
    <div
      class="medl__space"
    >
      <p
        class="typo-normal"
      >
        Kontakt Statens lånekasse for utdanning hvis du har spørsmål om perioder med støtte fra Lånekassen.
      </p>
    </div>
    <div
      class="medl__section"
    >
      <h2
        class="typo-systemtittel"
      >
        Oversikt over perioder
      </h2>
    </div>
    <div
      class="medl__space"
    >
      <div
        class="alertstripe alertstripe--info"
      >
        <span
          class="alertstripe__ikon"
        >
          <span
            class="sr-only"
          >
            info
          </span>
          <svg
            aria-label="info-ikon"
            focusable="false"
            height="1.5em"
            kind="info-sirkel-fyll"
            role="img"
            viewBox="0 0 24 24"
            width="1.5em"
          >
            <g
              fill="none"
            >
              <path
                d="M12 0C5.382 0 0 5.382 0 12s5.382 12 12 12c6.617 0 12-5.382 12-12S18.617 0 12 0z"
                fill="#337C9B"
              />
              <path
                d="M12 5a1.566 1.566 0 1 1 .11 3.13A1.566 1.566 0 0 1 12 5zm2.976 12.01c.563 0 1.043.431 1.043.991s-.48.992-1.043.992H9.39c-.564 0-1.043-.431-1.043-.992 0-.56.479-.99 1.043-.99h1.6v-5.016h-.986c-.565 0-1.044-.43-1.044-.991 0-.56.48-.991 1.044-.991h2.03c.563 0 1.043.43 1.043.99v6.007h1.899z"
                fill="#FFF"
              />
            </g>
          </svg>
        </span>
        <div
          class="typo-normal alertstripe__tekst"
        >
          Oversikten viser perioder fra og med 2015. Hvis perioden i 2015 henger sammen med en periode fra tidligere år, vil også den vises.
        </div>
      </div>
    </div>
    <div
      class="ekspanderbartPanel medl__space ekspanderbartPanel--lukket ekspanderbartPanel--border"
    >
      <button
        aria-controls="Mocked UUID"
        class="ekspanderbartPanel__hode"
        id="Mocked UUID"
        type="button"
      >
        <div
          class="ekspanderbartPanel__flex-wrapper"
        >
          <span
            class="ekspanderbartPanel__tittel"
          >
            Med medlemskap
          </span>
          <span
            class="ekspanderbartPanel__indikator"
          />
        </div>
      </button>
      <div
        aria-labelledby="Mocked UUID"
        id="Mocked UUID"
        role="region"
      />
    </div>
    <div
      class="ekspanderbartPanel medl__space ekspanderbartPanel--lukket ekspanderbartPanel--border"
    >
      <button
        aria-controls="Mocked UUID"
        class="ekspanderbartPanel__hode"
        id="Mocked UUID"
        type="button"
      >
        <div
          class="ekspanderbartPanel__flex-wrapper"
        >
          <span
            class="ekspanderbartPanel__tittel"
          >
            Uten medlemskap
          </span>
          <span
            class="ekspanderbartPanel__indikator"
          />
        </div>
      </button>
      <div
        aria-labelledby="Mocked UUID"
        id="Mocked UUID"
        role="region"
      />
    </div>
  </div>
</DocumentFragment>
`;
